### ISSUE TITLE
Preference Pane hangs on Snow Leopard

### ISSUE THREAD

**Issue by gabrielrinaldi** [Fingertips/passengerpane#8]
Passenger preference pane does not open on Snow Leopard. It hangs indefinitely.

**Comment by gf3** [comment]
Same here, I get the following messages in the console:

```
09-07-23 9:07:08 PM System Preferences[2458]    -[NSCFNumber pathForImageResource:]: unrecognized selector sent to instance 0x151cd690
09-07-23 9:07:08 PM [0x0-0x176176].com.apple.systempreferences  2009-07-23 21:07:08.277 System Preferences[2458:903] -[NSCFNumber pathForImageResource:]: unrecognized selector sent to instance 0x151cd690
09-07-23 9:07:08 PM System Preferences[2458]    -[NSCFNumber pathForImageResource:]: unrecognized selector sent to instance 0x151cd690
09-07-23 9:07:08 PM [0x0-0x176176].com.apple.systempreferences  2009-07-23 21:07:08.278 System Preferences[2458:903] -[NSCFNumber pathForImageResource:]: unrecognized selector sent to instance 0x151cd690
09-07-23 9:07:17 PM System Preferences[2458]    [NSPrefPaneBundle instantiatePrefPaneObject] (/Users/gianni/Library/PreferencePanes/Passenger.prefPane): should only be called once
09-07-23 9:07:17 PM [0x0-0x176176].com.apple.systempreferences  2009-07-23 21:07:17.865 System Preferences[2458:903] [NSPrefPaneBundle instantiatePrefPaneObject] (/Users/gianni/Library/PreferencePanes/Passenger.prefPane): should only be called once
09-07-23 9:07:18 PM System Preferences[2458]    PrefPanePassenger#willSelect: NoMethodError: undefined method `subviews' for nil:NilClass
09-07-23 9:07:18 PM [0x0-0x176176].com.apple.systempreferences  2009-07-23 21:07:18.260 System Preferences[2458:903] PrefPanePassenger#willSelect: NoMethodError: undefined method `subviews' for nil:NilClass
09-07-23 9:07:18 PM [0x0-0x176176].com.apple.systempreferences      /Users/gianni/Library/PreferencePanes/Passenger.prefPane/Contents/Resources/PassengerPref.rb:253:in `setup_passenger_warning!'
09-07-23 9:07:18 PM [0x0-0x176176].com.apple.systempreferences      /Users/gianni/Library/PreferencePanes/Passenger.prefPane/Contents/Resources/PassengerPref.rb:52:in `willSelect'
09-07-23 9:07:18 PM [0x0-0x176176].com.apple.systempreferences      /Users/gianni/Library/PreferencePanes/Passenger.prefPane/Contents/Resources/PassengerPref.rb:253
09-07-23 9:07:18 PM System Preferences[2458]    undefined method `subviews' for nil:NilClass
09-07-23 9:07:18 PM [0x0-0x176176].com.apple.systempreferences  2009-07-23 21:07:18.262 System Preferences[2458:903] undefined method `subviews' for nil:NilClass
```

**Comment by gabrielrinaldi** [comment]
You can fix this by installing RubyCocoa. You have to download it and fix the info.plist to make it install on SL.

**Comment by gf3** [comment]
Bah, I'm having separate issues with Passenger on SL anyways. Something about dlopen and a compiled bundle.

**Comment by rufo** [comment]
gianni, uninstall and re-install the passenger gem - it needs to be recompiled against 64-bit ruby and there's a native component that only compiles when you install the gem (I wound up recompiling my hand-built ruby and all my native gems, but I'm not sure you need to go that far.)

**Comment by nwp** [comment]
I'm getting this freezing on Snow Leopard as well.  I tried to install RubyCocoa but it won't let me even install it via the DMG file (it puts a ! on the drive and won't let you continue during the install), so that is a no go. I also re-installed passenger via the gem and the preference pane still doesn't work.

**Comment by jfernandez** [comment]
Somebody mentioned that you have to modify the DMG installer plist for Snow Leopard... yet they didn't mention what to do --- I've already googled all over the place and cant find anything yet.

**Comment by gabrielrinaldi** [comment]
You need to download RubyCocoa, copy the RubyCocoa-0.13.2-OSX10.5universal.pkg to your Desktop, right click it and choose Show Packages Contents. Then you open contents and on the Info.plist and inside IFRequirementDicts change versions to reflect Snow Leopard. That is all. After that passenger pane will work great!

**Comment by jfernandez** [comment]
Figured it out, see my blog post about it: http://tiny.cc/yo1qK

**Comment by shanebo** [comment]
@jfernandez I followed your directions and got RubyCocoa installed. Thanks! Then I uninstalled the passenger gem and attempted to reinstall the passenger gem and got:

ERROR:  Error installing passenger:
    ERROR: Failed to build gem native extension.

/System/Library/Frameworks/Ruby.framework/Versions/1.8/usr/bin/ruby extconf.rb
mkmf.rb can't find header files for ruby at /System/Library/Frameworks/Ruby.framework/Versions/1.8/usr/lib/ruby/ruby.h

Gem files will remain installed in /Library/Ruby/Gems/1.8/gems/passenger-2.2.4 for inspection.
Results logged to /Library/Ruby/Gems/1.8/gems/passenger-2.2.4/ext/phusion_passenger/gem_make.out

---

Did you run into this? Any idea how to resolve this? Any help is appreciated.

**Comment by bensie** [comment]
@jfernandez, thanks your blog solved the issue for me.

**Comment by Manfred** [comment]
@shanebo, it means you don't have the Ruby headers. You have to re-install Xcode.

**Comment by jrmehle** [comment]
@jrfernandez: The blog post helped me get the preference pane to open again, but now my problem seems to be I can't reach any of the local domains I've set up in it.

Update: I had to delete and re-add all of my projects in the preference pane, but they do come up in the browser now.

**Comment by jfernandez** [comment]
Run the passenger apache installer again and setup the passenger conf (snow leopard nukes your httpd.conf)

**Comment by shanebo** [comment]
@jfernandez I did. Here's what I've installed since the Snow Leopard install.

RubyCocoa 13.2 > xcode > passenger 2.2.4 > apache2 module along with the 3 config lines in the conf > passenger pref pane 1.2

All my existing vhosts _do_ exist but can't access them via localhost. I've fired up a ramaze app via terminal to make sure it's not a ruby and/or ramaze issue and all is good in that regard so it must be a passenger problem... Any other ideas?

**Comment by plainprogrammer** [comment]
Just cloned the project and brought it up in XCode 3.2 on Snow Leopard. Built with default options that came up: OS X 10.5 | Debug | i386

Got the following mess in Console upon putting it into System Preferences:

```
8/30/09 8:54:44 PM  System Preferences[9951]    -[NSCFNumber pathForImageResource:]: unrecognized selector sent to instance 0x15993f70
8/30/09 8:54:44 PM  [0x0-0xce0ce].com.apple.systempreferences   2009-08-30 20:54:44.373 System Preferences[9951:903] -[NSCFNumber pathForImageResource:]: unrecognized selector sent to instance 0x15993f70
8/30/09 8:54:49 PM  System Preferences[9951]    [NSPrefPaneBundle instantiatePrefPaneObject] (/Users/jamesthompson/Library/PreferencePanes/Passenger.prefPane): should only be called once
8/30/09 8:54:49 PM  [0x0-0xce0ce].com.apple.systempreferences   2009-08-30 20:54:49.753 System Preferences[9951:903] [NSPrefPaneBundle instantiatePrefPaneObject] (/Users/jamesthompson/Library/PreferencePanes/Passenger.prefPane): should only be called once
8/30/09 8:54:49 PM  System Preferences[9951]    PrefPanePassenger#willSelect: NoMethodError: undefined method `subviews' for nil:NilClass
8/30/09 8:54:49 PM  [0x0-0xce0ce].com.apple.systempreferences   2009-08-30 20:54:49.818 System Preferences[9951:903] PrefPanePassenger#willSelect: NoMethodError: undefined method `subviews' for nil:NilClass
8/30/09 8:54:49 PM  [0x0-0xce0ce].com.apple.systempreferences       /Users/jamesthompson/Library/PreferencePanes/Passenger.prefPane/Contents/Resources/PassengerPref.rb:253:in `setup_passenger_warning!'
8/30/09 8:54:49 PM  [0x0-0xce0ce].com.apple.systempreferences       /Users/jamesthompson/Library/PreferencePanes/Passenger.prefPane/Contents/Resources/PassengerPref.rb:52:in `willSelect'
8/30/09 8:54:49 PM  System Preferences[9951]    undefined method `subviews' for nil:NilClass
8/30/09 8:54:49 PM  [0x0-0xce0ce].com.apple.systempreferences   2009-08-30 20:54:49.819 System Preferences[9951:903] undefined method `subviews' for nil:NilClass
```

I tried fiddling with the build a bit, moving it to use 10.6 SDK, and x86_64 architecture and the errors change to being concerned with garbage collection. So there may be a number of issues at work in getting this working on Snow Leopard.

**Comment by mikew** [comment]
I'm getting a different error, but from the same method:

```
PrefPanePassenger#willSelect: NoMethodError: undefined method `hidden=' for nil:NilClass
/Library/PreferencePanes/Passenger.prefPane/Contents/Resources/PassengerPref.rb:250:in `setup_passenger_warning!'
/Library/PreferencePanes/Passenger.prefPane/Contents/Resources/PassengerPref.rb:52:in `willSelect'
```

**Comment by Manfred** [comment]
No worries people, we know about the problem and we will fix it. Eloy has been under the weather an didn't have time to look at it yet. Please hang in there for a while. In the meantime patches are always welcome.

**Comment by alloy** [comment]
Added RubyCocoa 0.13.2 as compiled for 10.5 universal to vendor and made the project link against and bundle this version.

Closed by 26e905a9afce7196d4db3e56f356be8129bb2309.
